tools/tokenize.lua

```diff
@@ -15,7 +15,12 @@ cmd:option('-joiner_annotate', false, [[Include joiner annotation using 'joiner'
 cmd:option('-joiner', separators.joiner_marker, [[Character used to annotate joiners]])
 cmd:option('-joiner_new', false, [[in joiner_annotate mode, 'joiner' is an independent token]])
 cmd:option('-case_feature', false, [[Generate case feature]])
-cmd:option('-bpe_model', '', [[Apply Byte Pair Encoding if the BPE model path is given]])
+cmd:option('-bpe_model', '', [[Apply Byte Pair Encoding if the BPE model path is given. If the option is used, 'mode' will be overridden/set automatically if the BPE model specified by bpe_model is learnt using learn_bpe.lua]])
+cmd:option('-EOT_marker', separators.EOT, [[Marker used to mark the end of token, use '</w>' for python models, otherwise default value ]])
+cmd:option('-BOT_marker', separators.BOT, [[Marker used to mark the begining of token]])
+cmd:option('-bpe_case_insensitive', false, [[Apply BPE internally in lowercase, but still output the truecase units. This option will be overridden/set automatically if the BPE model specified by bpe_model is learnt using learn_bpe.lua]])
```

Comment:
For those who use models learnt by learn_bpe.py on a lowercased training set with their own tokenization